flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/sink/shuffle/TestSketchUtil.java

```diff
@@ -130,4 +133,111 @@ public void testRangeBoundsSkipDuplicates() {
         // skipped duplicate c's
         .containsExactly(CHAR_KEYS.get("c"), CHAR_KEYS.get("g"), CHAR_KEYS.get("j"));
   }
+
+  @ParameterizedTest
+  @ValueSource(ints = {4, 6})
+  public void testPartitioningAndScaleUp(int numPartitions) {
+    // calculated based on 4 partitions
+    SortKey[] rangeBounds =
+        new SortKey[] {CHAR_KEYS.get("c"), CHAR_KEYS.get("j"), CHAR_KEYS.get("m")};
+
+    // <= c
+    assertThat(
```

Comment:
Is this code duplicated in the next method? Is there a way to extract the duplicated code?

Reply:
there are some duplications. but I think those two tests are different: scale up and scale down. I want to explicitly test each range.

The scale down case has this differently.
```
// > m
    // reassigns out-of-range partitions via mod (% 3 in this case)
    assertThat(
            SketchUtil.partition(
                CHAR_KEYS.get("n"), numPartitions, rangeBounds, SORT_ORDER_COMPARTOR))
        .isEqualTo(0);
    assertThat(
            SketchUtil.partition(
                CHAR_KEYS.get("z"), numPartitions, rangeBounds, SORT_ORDER_COMPARTOR))
        .isEqualTo(0);
```

Reply:
Maybe a method like this would help to focus on the actual test and hide the common parameters?
```
private static assertPartition(SortKey key, int expected) {
   assertThat(
            SketchUtil.partition(
                key, numPartitions, rangeBounds, SORT_ORDER_COMPARTOR))
        .isEqualTo(expected);
}
```

The first 8 assertions seems like the same to me, but it is hard to check/understand as it is "hidden" by the long assertion statements.

WDYT?

Reply:
that is a good idea. Reducing to a single line would help readability.